File: src/main/java/by/intexsoft/course/model/Hotel.java
package by.intexsoft.course.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

/**
 * Model for hotels table
 */
@Entity
@Table(name = "hotels")
public class Hotel extends AbstractModel {

	private static final long serialVersionUID = -6108533233304560675L;

	/**
	 * Name of hotel
	 */
	@Column
	public String name;

	/**
	 * Address of hotel
	 */
	@Column
	public String address;

	/**
	 * {@link Town} in which the hotel is located
	 */
	@ManyToOne(targetEntity = Town.class, fetch = FetchType.EAGER)
	public Town town;
}

File: client/src/app/model/hotel.ts
import {Town} from "./town";
export class Hotel {
    id: number;
    name: string;
    address: string;
    town: Town;

    constructor(name: string, address: string, town: Town) {
        this.name = name;
        this.address = address;
        this.town = town
    }
}